nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutSQL.java

```diff
@@ -828,10 +833,50 @@ private void setParameter(final PreparedStatement stmt, final String attrName, f
                     stmt.setBigDecimal(parameterIndex, new BigDecimal(parameterValue));
                     break;
                 case Types.DATE:
-                    stmt.setDate(parameterIndex, new Date(Long.parseLong(parameterValue)));
+                    Date date;
+
+                    if (valueFormat.equals("")) {
+                        if(LONG_PATTERN.matcher(parameterValue).matches()){
+                            date = new Date(Long.parseLong(parameterValue));
+                        }else {
```

Comment:
We prefer to have a space here, like `} else {`. The existing code for TIMESTAMP has the similar code, too.